FHEM/00_SIGNALduino.pm

```diff
@@ -728,7 +731,7 @@ sub SIGNALduino_Attr_rfmode {
   my $hash = shift // carp 'must be called with hash of iodevice as first param';
   my $aVal = shift // return;
 
-  if ( (InternalVal($hash->{NAME},"cc1101_available",0) == 0) && (!IsDummy($hash->{NAME})) ) {
+  if ( ( (not defined InternalVal($hash->{NAME},"cc1101_available",undef)) || (InternalVal($hash->{NAME},"cc1101_available",0) == 0) ) && (!IsDummy($hash->{NAME})) ) {
```

Comment:
Das sieht mir irgendwie nach doppelt hält besser aus.

InternalVal gibt doch den dritten Parameter zurück, wenn das Internal nicht definiert ist. Den Zweck es doppelt zu erkenne sehe ich nicht.

Reply:
Wir haben es so gelöst weil es Fälle gab wo es durchfiel sonst. Doppelt war es meines Erachtens nicht. An die genaue Testconfig müsste ich versuchen mich zu erinnern. Da war was... Ich weiß nur, das wir es auch von einem Device ohne cc1101 probierten und von Dummy glaube aus.

Reply:
Dann müsste diese besondere Testconfig in dem Test berücksichtigt werden.
Es wird der 3. Parameter zurückgeliefert, wenn es das Internal nicht gibt.
Das ist dann undef bzw 0.  Zu der 2. Prüfung kommt es natürlich nie, solange das Attribut nicht definiert ist.
Erst, wenn es mit dem Wert 0 definiert wurde, dann wird die 2. Abfrage gestartet. Alles in allem, kann ich mir nicht vorstellen, dass die vorherige Variante nicht wie gewünscht funktioniert hat :(

Reply:
Den genau Fall kann ich erst nächste Woche erneut testen. Vielleicht kann sich ja @elektron-bbs noch dran erinnern. Ansonsten müssen wir warten oder ich versuche es mal mit den hier mir zu Verfügung stehenden Mitteln zu reproduzieren.

Reply:
Das würde mich sehr interessieren denn, cc1101_available kann vom Prinzip drei Werte haben

1:  undef   , dann gibt `(InternalVal($hash->{NAME},"cc1101_available",0)` eine 0 zurück
2:  0  , dann gibt `(InternalVal($hash->{NAME},"cc1101_available",0)` eine 0 zurück
2:  1  , dann gibt `(InternalVal($hash->{NAME},"cc1101_available",0)` eine 1 zurück

Alle anderen Werte, könnte man theoretisch manuell da reinschreiben, aber dann würde eben genau dieser Wert zurück geliefert werden.

Reply:
Ich denke, @sidey79 liegt richtig.
Solange $ret in SIGNALduino_Attr nicht ausgewertet wurde, wurde das Attribut rfmode immer gesetzt, egal was beim return zurück gegeben wurde.

Reply:
@sidey79, ich habe noch einmal geschaut und ihr habt natürlich Recht.
Ich habe da was durcheinader gebracht und habe den doppelten Syntax abgeändert. https://github.com/RFD-FHEM/RFFHEM/pull/905/commits/768e31834997a371d8f66b52d501c442ca742525